### PR TITLE
refactor!: `ProgressSegmented` replace with experimental component, add migration

### PR DESCRIPTION
**Pull request by vladimirpotekhin**
part of #5808

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/demo/src/modules/directives/progress-segmented/index.html [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <tui-doc-page
-    header="ProgressSegmented "
-    package="EXPERIMENTAL"
+    header="ProgressSegmented"
+    package="KIT"
     type="directives"
```
Comment (nsbarsukov) [inline projects/demo/src/modules/directives/progress-segmented/index.html:4]: Снимок экрана 2024-03-26 в 12 13 10

File: projects/demo/src/modules/directives/progress-segmented/index.html [hunk 2]
```diff
@@ -160,31 +128,5 @@
...
-                <p>Add to the template:</p>
-
-                <tui-doc-code
-                    filename="my.component.html"
-                    [code]="exampleHtml"
-                />
-            </li>
-        </ol>
-    </ng-template>
+    <tui-setup *pageTab />
```
Comment (nsbarsukov) [inline projects/demo/src/modules/directives/progress-segmented/index.html:131]: Empty ?
Снимок экрана 2024-03-26 в 12 15 02

File: projects/kit/components/progress/progress-bar/progress-bar.component.less [hunk 3]
```diff
@@ -41,62 +41,43 @@
...
     }
 
     &[data-size='s'] {
-        .progressValue({ border-radius: var(--tui-radius-xs) });
-
         --t-height: 0.5rem;
     }
 
-    // TODO: drop new attribute in v4.0 and make all these styles as default ones
-    &[new] {
```
Comment (nsbarsukov) [inline projects/kit/components/progress/progress-bar/progress-bar.component.less:69]: Update this documentation page

https://taiga-ui--pr7092-progress-move-jtxz5ldx.web.app/components/progress-bar#sizes

Comment (nsbarsukov) [reply projects/kit/components/progress/progress-bar/progress-bar.component.less:69]: And this example

https://taiga-ui--pr7092-progress-move-jtxz5ldx.web.app/components/progress-bar#customizable-corners

Comment (nsbarsukov) [reply projects/kit/components/progress/progress-bar/progress-bar.component.less:69]: Let's use bigger size for this example:

https://taiga-ui--pr7092-progress-move-jtxz5ldx.web.app/components/progress-bar#label

**Review by nsbarsukov — APPROVED** [review]

**Pull request merged**